connector/src/main/java/org/geysermc/connector/network/translators/java/entity/spawn/JavaSpawnEntityTranslator.java

```diff
@@ -69,8 +70,17 @@ public void translate(ServerSpawnEntityPacket packet, GeyserSession session) {
                         type, position, motion, rotation, (HangingDirection) packet.getData());
             } else if (packet.getType() == EntityType.FISHING_BOBBER) {
                 // Fishing bobbers need the owner for the line
-                entity = new FishingHookEntity(packet.getEntityId(), session.getEntityCache().getNextEntityId().incrementAndGet(),
-                        type, position, motion, rotation, (ProjectileData) packet.getData());
+                int ownerEntityId = ((ProjectileData) packet.getData()).getOwnerId();
+                Entity owner = session.getEntityCache().getEntityByJavaId(ownerEntityId);
+                if (owner == null && session.getPlayerEntity().getEntityId() == ownerEntityId) {
+                    owner = session.getPlayerEntity();
+                }
+                if (owner instanceof PlayerEntity) {
+                    entity = new FishingHookEntity(packet.getEntityId(), session.getEntityCache().getNextEntityId().incrementAndGet(),
+                            type, position, motion, rotation, (PlayerEntity) owner);
+                } else {
+                    return;
```

Comment:
Can a fishing hook entity not spawn if the owner isn't a player? If that is correct Java behavior, then I ask that you add a comment explaining this.